### DISCUSSION TITLE
Two entries per change

### DISCUSSION THREAD

**Discussion by wiwa1978** [Z3d0X/filament-logger#108] [Q&A]
For every change I make to a model, it gets added twice to the activity-log table and hence also showed twice on the UI. Any idea why this is? I basically just installed spatie/laravel-activitylog and then the filament-logger package.

Note: this is only for updates to resource models. Notifications and Access type are only logged once.

![image](https://github.com/Z3d0X/filament-logger/assets/4012455/bfaab77c-ba6c-412e-96b4-0f92fd26b5a9)

**Comment by DonatasIk** [comment]
Seems like the issue occurs because multiple resources are tied to the same model. I had this problem, and while my solution most likely is not the best one, but at least it works for my use case.

Create a new class `app/Loggers/CustomResourceLogger.php`:

```
namespace App\Loggers;

use Illuminate\Database\Eloquent\Model;
use Illuminate\Support\Facades\Cache;
use Z3d0X\FilamentLogger\Loggers\AbstractModelLogger;

class CustomResourceLogger extends AbstractModelLogger
{
    protected function getLogName(): string
    {
        return config('filament-logger.resources.log_name');
    }

    protected function getCacheKey(Model $model, string $event): string
    {
        return 'filament_log_' . $model::class . '_' . $model->getKey() . '_' . $event . '_' . now()->format('Y-m-d_H:i:s');
    }

    protected function log(Model $model, string $event, ?string $description = null, mixed $attributes = null): void
    {
        $cacheKey = $this->getCacheKey($model, $event);

        $shouldLog = Cache::lock($cacheKey . '_lock', 10)->get(function () use ($cacheKey) {
            if (Cache::has($cacheKey)) {
                return false;
            }

            Cache::put($cacheKey, true, now()->addSeconds(5));

            return true;
        });

        if ($shouldLog) {
            parent::log($model, $event, $description, $attributes);
        }
    }
}
```

In the `config/filament-logger.php` file, update the resource logger to use this custom class:

```
'resources' => [
    ...
    'logger' => \App\Loggers\CustomResourceLogger::class,
    ...
]
```
